pkg/controllers/manifestwork.go

```diff
@@ -160,6 +160,28 @@ func syncManifestworkStatusToHypershiftDeployment(
 	}
 }
 
+func (r *HypershiftDeploymentReconciler) validateHostedClusterAndNodePool(ctx context.Context, hcName string, hcSpec hyp.HostedClusterSpec, npSpec hyp.NodePoolSpec) error {
+	// Platform.Type in NodePool matches the HostedCluster
+	if npSpec.Platform.Type != hcSpec.Platform.Type {
+		r.Log.Error(errors.New("Platform.Type value mismatch"), "Platform.Type in node pool(s) does not match value in HostedClusterSpec")
+		return errors.New("Platform.Type value mismatch")
+	}
+
+	// NodePool references the correct hostedCluster
+	if npSpec.ClusterName != hcName {
+		r.Log.Error(errors.New("incorrect Spec.ClusterName in NodePool"), "Spec.ClusterName in NodePool needs to match the referenced hostedCluster")
+		return errors.New("incorrect Spec.ClusterName in NodePool")
+	}
+
+	// Release.Image in NodePool matches the HostedCluster
+	if npSpec.Release.Image != hcSpec.Release.Image {
```

Comment:
So this is allowed, though I'm not sure what is legal and what isn't.

ie. Even at provisioning, there could be a case to run NodePool as 4.11 and hostedCluster as 4.10.  As well during an upgrade, you might see the hostedCluster move before the nodepool or vice versa, though I believe this check is only on provisioning... So not sure what we want to do, but I'm not sure we should block it.

Reply:
After speaking with @philipwu08 he will leave it as a warning.
/lgtm